packages/ckeditor5-restricted-editing/src/restrictededitingmodeui.ts

```diff
@@ -52,6 +52,7 @@ export default class RestrictedEditingModeUI extends Plugin {
 			addListToDropdown( dropdownView, listItems );
 
 			dropdownView.buttonView.set( {
+				role: 'menu',
```

Comment:
As discussed internally, for the `role=menu` to make sense, this should be set on the `ListView` instance instead. `addListToDropdown()` offers options configuration to set the role but can also be set manually.